pkg/blob/controllerserver.go

```diff
@@ -768,18 +774,75 @@ func (d *Driver) copyBlobContainer(_ context.Context, req *csi.CreateVolumeReque
 }
 
 // copyVolume copies a volume form volume or snapshot, snapshot is not supported now
-func (d *Driver) copyVolume(ctx context.Context, req *csi.CreateVolumeRequest, accountKey, dstContainerName, storageEndpointSuffix string) error {
+func (d *Driver) copyVolume(ctx context.Context, req *csi.CreateVolumeRequest, accountSASToken, dstContainerName, storageEndpointSuffix string) error {
 	vs := req.VolumeContentSource
 	switch vs.Type.(type) {
 	case *csi.VolumeContentSource_Snapshot:
 		return status.Errorf(codes.InvalidArgument, "copy volume from volumeSnapshot is not supported")
 	case *csi.VolumeContentSource_Volume:
-		return d.copyBlobContainer(ctx, req, accountKey, dstContainerName, storageEndpointSuffix)
+		return d.copyBlobContainer(ctx, req, accountSASToken, dstContainerName, storageEndpointSuffix)
 	default:
 		return status.Errorf(codes.InvalidArgument, "%v is not a proper volume source", vs)
 	}
 }
 
+func (d *Driver) authorizeAzcopyWithIdentity() error {
+	azureAuthConfig := d.cloud.Config.AzureAuthConfig
+	if len(azureAuthConfig.AADClientSecret) > 0 {
+		klog.V(2).Infof("use service principal to authorize azcopy")
+		if err := os.Setenv(azcopyAutoLoginType, "SPN"); err != nil {
+			return err
+		}
+		if azureAuthConfig.AADClientID == "" {
+			return fmt.Errorf("AADClientID and AADClientSecret must be set when use service principal")
+		}
+		if err := os.Setenv(azcopySPAApplicationID, azureAuthConfig.AADClientID); err != nil {
+			return err
+		}
+		if err := os.Setenv(azcopySPAClientSecret, azureAuthConfig.AADClientSecret); err != nil {
```

Comment:
`os.Setenv` set the var in the driver, use `cmd.Env` is more secure

https://github.com/kubernetes-sigs/blob-csi-driver/blob/a167f2e59b9eb797119d4e9d1e704a04d5272988/pkg/blob/nodeserver.go#L190

Reply:
done, transfer authAzcopyEnv for cmd.Env